File: src-scala/org/jgll/grammar/slotaction/SlotAction.scala
package org.jgll.grammar.slotaction

import java.io.Serializable
import org.jgll.parser.GLLParserInternalsTrait
import org.jgll.util.InputTrait
//remove if not needed

trait SlotActionTrait { self: InputTrait with GLLParserInternalsTrait =>
  trait SlotAction[T] extends Serializable {

    def execute(parser: GLLParserInternals, input: Input): T
  }
}

File: java/src/org/jgll/grammar/slotaction/SlotAction.java
package org.jgll.grammar.slotaction;

import java.io.Serializable;

import org.jgll.parser.GLLParserInternals;
import org.jgll.util.Input;

public interface SlotAction<T> extends Serializable {
	
	public T execute(GLLParserInternals parser, Input input);
	
}
